server/game/cards/05-DT/SelectivePreservation.js

```diff
@@ -3,20 +3,44 @@ const Card = require('../../Card.js');
 class SelectivePreservation extends Card {
     setupCardAbilities(ability) {
         this.play({
-            target: {
-                activePromptTitle: 'Choose a creature of each power value to not destroy',
-                cardType: 'creature',
-                mode: 'exactly',
-                numCards: (context) =>
-                    new Set(context.game.creaturesInPlay.map((card) => card.power)).size,
-                selectorCondition: (selectedCards, context) =>
-                    new Set(context.game.creaturesInPlay.map((card) => card.power)).size ===
-                    new Set(selectedCards.map((card) => card.power)).size,
-                gameAction: ability.actions.destroy((context) => ({
-                    target: context.game.creaturesInPlay.filter(
-                        (card) => context.target && !context.target.includes(card)
+            effect: 'Select creatures for each power to not destroy.',
+
+            then: (preThenContext) => {
+                let unqiuePowers = Array.from(
+                    new Set(
+                        preThenContext.game.creaturesInPlay.map((creature) =>
+                            creature.getPower(false)
+                        )
                     )
-                }))
+                );
+                unqiuePowers.sort((a, b) => (a > b ? 1 : -1));
+
+                let targets = [];
+                for (let i = 0; i < unqiuePowers.length; i++) {
+                    let power = unqiuePowers[i];
+                    let targetKey = 'power' + power;
+                    targets[targetKey] = {
+                        activePromptTitle: {
+                            text: 'Choose a {{power}} creature to not destroy',
+                            values: { power: power }
+                        },
+                        cardType: 'creature',
+                        numCards: 1,
+                        cardCondition: (card) => card.getPower(false) === power,
+                        gameAction: ability.actions.destroy((context) => ({
```

Comment:
I believe the gameAction needs to be outside all targets and you need to make sure the card is not in any target selection